crates/orderbook/src/orderbook.rs

```diff
@@ -206,6 +207,11 @@ impl Orderbook {
     ///
     /// The full app data can be located in the database or on IPFS.
     pub async fn find_full_app_data(&self, contract_app_data: &AppDataHash) -> Option<String> {
+        // we reserve the 0 app data to indicate empty app data.
+        if contract_app_data.is_zero() {
+            return Some(app_data::EMPTY.to_string());
+        }
```

Comment:
Is not needed because I added database entries for all zero app data mapping to `{}`. On the other hand if you start with a fresh database like in an e2e test that would stop working.

Reply:
Ah ok - makes sense. This is is why I ran into issues locally.

I would prefer to keep this as explicit behaviour instead of relying on the DB being set up (unless it became part of the migration process).